### PR TITLE
UGENE-7141. Remove trivial GCC warnings suppression

### PR DESCRIPTION
**Pull request by yalgaer**
The final goal is to remove suppression for all warnings, so UGENE will be warning free and our GCC-warning-check build  in the Teamcity can alert when new warnings are found.

The warnings that left after this PR require UGENE code modification and will be done in dedicated PRs.

### CODE REVIEW

**Review by evelinabis — APPROVED** [review]

File: src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignment.cpp [hunk 1]
```diff
@@ -604,7 +604,7 @@ MultipleChromatogramAlignment MultipleChromatogramAlignmentData::getExplicitCopy
 void MultipleChromatogramAlignmentData::copy(const MultipleAlignmentData &other) {
     try {
         copy(dynamic_cast<const MultipleChromatogramAlignmentData &>(other));
-    } catch (std::bad_cast) {
+    } catch (std::bad_cast &) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignment.cpp:607]: I suppose, it's better to catch `const std::bad_cast &`

Comment (yalgaer) [reply src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignment.cpp:607]: I also do not like the code here, but behavior changes are out of scope in this PR

**Review by DmitriiSukhomlinov — APPROVED** [review]
Minor comments

File: src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignment.cpp (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignment.cpp:607]: In this case, why did you change `std::bad_cast` to `std::bad_cast &`?

Comment (yalgaer) [reply src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignment.cpp:607]: The warning (-Wcatch-value) was not related to the empty catch clause but to the way the value is caught. See other places with non-empty catch clause that were also changed .

**Pull request merged**